lib/git_helper/merge_request.rb

```diff
@@ -101,7 +101,20 @@ def merge
     end
 
     private def default_branch
-      @default_branch ||= gitlab_client.branches(local_repo).select { |branch| branch.default }.first.name
+      return @default_branch if @default_branch
+      page_number = 1
+      counter = 1
+      branches = []
+
+      while counter > 0
+        break if default_branch = branches.select { |branch| branch.default }.first
+        page_branches = gitlab_client.branches(local_project, page: page_number, per_page: 100)
+        branches = page_branches
+        counter = page_branches.count
+        page_number += 1
+      end
+
+      @default_branch = default_branch.name
```

Comment:
Technically, there is a blank space in the logic where if there was no default branch, the code will loop all the way through _all_ pages of the branches, and not come out with a `default_branch` defined. In that way, this line would throw an error that it couldn't find `name` for `nil`. However, I don't know if that would realistically happen, since I don't think you could have a project w/out a default branch. Fault in the API maybe.